lms/views/basic_lti_launch.py

```diff
@@ -159,49 +159,23 @@ def unconfigured_basic_lti_launch(self):
         we'll save it in our DB. Subsequent launches of the same assignment
         will then be DB-configured launches rather than unconfigured.
         """
-        self.context.js_config.enable_content_item_selection_mode()
-
-        params = self._extract_lti_params(self.request)
+        form_fields = {
+            param: value
+            for param, value in self.request.params.items()
+            if param not in ["oauth_nonce", "oauth_timestamp", "oauth_signature"]
+        }
```

Comment:
I think in-lining this here is really good. It stops you having to immediately jump to another function upon reading this one.

It's possible to do this without this becoming overwhelming because so much has been extracted below. At this point you could probably get away with just using this dict in the call to `enable_content_item_selection_mode()` as we do in the `content_item_selection` view.

The comment below about why we are removing these particular parameters was helpful though (because they aren't re-usable)